scrapy_poet/page_input_providers.py

```diff
@@ -329,6 +344,20 @@ def update_cache(self, request: Request, mapping: Dict[Type, Any]) -> None:
     def get_from_cache(self, request: Request, cls: Callable) -> Optional[Any]:
         return self._cached_instances.get(request, {}).get(cls)
 
+    def check_if_deadlock(self, request: Request) -> bool:
+        """Should only be used when ``AsyncioSelectorReactor`` is the reactor."""
+        if request not in self._build_instances_call_counter:
+            self._build_instances_call_counter[request] = 0
+        self._build_instances_call_counter[request] += 1
```

Comment:
I've thought about storing the `andi.Plan` here and checking how many times we've seen it before to see if there's a potential deadlock. However, it take up a lot of memory on some cases.

Tracking and storing a simple `int` value like this per `scrapy.Request` is much lighter on the memory.